cmd/common/transaction.go

```diff
@@ -237,7 +238,15 @@ func SignParaTimeTransaction(
 		tx.Call = *encCall
 	}
 
-	PrintTransactionBeforeSigning(npa, tx)
+	PrintTransactionBeforeSigning(npa, &types.PrettyTransaction{
+		Versioned: tx.Versioned,
+		Call: types.PrettyCall{
+			Format: tx.Call.Format,
+			Method: tx.Call.Method,
+			Body:   txBody,
```

Comment:
Why does it need to get the `txBody` like this and what if one passes something that is not actually consistent with the `tx`?

I suggest we implement a map from `module.MethodName` to tx type and then use that for pretty printing (e.g. similar to what we do in Core).

Reply:
Implemented in oasisprotocol/oasis-sdk#1269.